src/Ng/Program.cs

```diff
@@ -27,22 +27,22 @@ public static void Main(string[] args)
                 Debugger.Launch();
             }
 
-            // Get arguments
-            var arguments = CommandHelpers.GetArguments(args, 0);
-
-            // Configure ApplicationInsights
-            ApplicationInsights.Initialize(CommandHelpers.GetApplicationInsightsInstrumentationKey(arguments));
-
             // Create an ILoggerFactory
             var loggerConfiguration = LoggingSetup.CreateDefaultLoggerConfiguration(withConsoleLogger: true);
             var loggerFactory = LoggingSetup.CreateLoggerFactory(loggerConfiguration);
 
             // Create a logger that is scoped to this class (only)
             _logger = loggerFactory.CreateLogger<Program>();
 
-            var cancellationTokenSource = new CancellationTokenSource();
             try
             {
+                // Get arguments
+                var arguments = CommandHelpers.GetArguments(args, 0);
+
+                // Configure ApplicationInsights
+                ApplicationInsights.Initialize(CommandHelpers.GetApplicationInsightsInstrumentationKey(arguments));
```

Comment:
I have moved a few things around and put this back where it was before after app insights initialization.

---
In reply to: [71253386](https://github.com/NuGet/NuGet.Services.Metadata/pull/77#discussion_r71253386) [](ancestors = 71253386)